### ISSUE TITLE
Should we provide a config that disables some rules from `@typescript-eslint/eslint-plugin`?

### ISSUE THREAD

**Issue by NullVoxPopuli** [ember-cli/eslint-plugin-ember#2110]
For example, no-use-before-define is often wrong if you define multiple components in the same file.

```gjs
const Foo = <template><Bar /></template>;
const Bar = <template>hi</template>;
```

would trigger the error.
But this is totally valid, because component rendering is not _eager_ like const-evaluation normally is.

cc @patricklx, in case you have ideas

**Comment by patricklx** [comment]
this? https://github.com/ember-cli/eslint-plugin-ember/pull/2107

**Comment by NullVoxPopuli** [comment]
> mmm, this might be different .

yeah, that PR adds rules to a config, which I don't think we should do.

I imagine the TS config should end up looking like this:
```js
{
  // this is the exact same as what's in the README
  files: ['**/*.gts'],
  parser: 'ember-eslint-parser',
  plugins: ['ember'],
  extends: [
    'eslint:recommended',
    // hopefully this either adds the rules to the config (so that 2107 isn't needed)
    // or we give up on legacy configs and only recommend flat config
    'plugin:@typescript-eslint/recommended',
    'plugin:ember/recommended',
     // this would detect if @typescript-eslint is present, 
     // and make sure no-use-before-define is turned off
    'plugin:ember/recommended-gts', 
  ],
},
```

**Comment by patricklx** [comment]
I think the error is correct though.
 it wil be
```
Foo =  precompileTemplatr(`...`,  {
  scope: [Bar]
})
```

**Comment by patricklx** [comment]
I think it was pretty clear that they will not change `plugin:@typescript-eslint/recommended` behaviour. (Lets discuss this part in #2107)

**Comment by patricklx** [comment]
> > it wilz be compiled to something like
> 
> nay, it's
> 
> ```
> scope: () => ({ Bar })
> ```

hmm, i took another look, you are right:)

But it looks like the rule isn't looking that exactly:
[eslint playground](
https://eslint.org/play/#eyJ0ZXh0IjoicHJlY29tcGlsZSgnJywge1xuICBzY29wZTogKCkgPT4gW0Jhcl1cbn0pXG5cbmNvbnN0IEJhciA9IDE7Iiwib3B0aW9ucyI6eyJlbnYiOnsiZXM2Ijp0cnVlfSwicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9LCJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQifSwicnVsZXMiOnsiY29uc3RydWN0b3Itc3VwZXIiOlsiZXJyb3IiXSwiZm9yLWRpcmVjdGlvbiI6WyJlcnJvciJdLCJnZXR0ZXItcmV0dXJuIjpbImVycm9yIl0sIm5vLWFzeW5jLXByb21pc2UtZXhlY3V0b3IiOlsiZXJyb3IiXSwibm8tY2FzZS1kZWNsYXJhdGlvbnMiOlsiZXJyb3IiXSwibm8tY2xhc3MtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbXBhcmUtbmVnLXplcm8iOlsiZXJyb3IiXSwibm8tY29uZC1hc3NpZ24iOlsiZXJyb3IiXSwibm8tY29uc3QtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbnN0YW50LWNvbmRpdGlvbiI6WyJlcnJvciJdLCJuby1jb250cm9sLXJlZ2V4IjpbImVycm9yIl0sIm5vLWRlYnVnZ2VyIjpbImVycm9yIl0sIm5vLWRlbGV0ZS12YXIiOlsiZXJyb3IiXSwibm8tZHVwZS1hcmdzIjpbImVycm9yIl0sIm5vLWR1cGUtY2xhc3MtbWVtYmVycyI6WyJlcnJvciJdLCJuby1kdXBlLWVsc2UtaWYiOlsiZXJyb3IiXSwibm8tZHVwZS1rZXlzIjpbImVycm9yIl0sIm5vLWR1cGxpY2F0ZS1jYXNlIjpbImVycm9yIl0sIm5vLWVtcHR5IjpbImVycm9yIl0sIm5vLWVtcHR5LWNoYXJhY3Rlci1jbGFzcyI6WyJlcnJvciJdLCJuby1lbXB0eS1wYXR0ZXJuIjpbImVycm9yIl0sIm5vLWV4LWFzc2lnbiI6WyJlcnJvciJdLCJuby1leHRyYS1ib29sZWFuLWNhc3QiOlsiZXJyb3IiXSwibm8tZXh0cmEtc2VtaSI6WyJlcnJvciJdLCJuby1mYWxsdGhyb3VnaCI6WyJlcnJvciJdLCJuby1mdW5jLWFzc2lnbiI6WyJlcnJvciJdLCJuby1nbG9iYWwtYXNzaWduIjpbImVycm9yIl0sIm5vLWltcG9ydC1hc3NpZ24iOlsiZXJyb3IiXSwibm8taW5uZXItZGVjbGFyYXRpb25zIjpbImVycm9yIl0sIm5vLWludmFsaWQtcmVnZXhwIjpbImVycm9yIl0sIm5vLWlycmVndWxhci13aGl0ZXNwYWNlIjpbImVycm9yIl0sIm5vLWxvc3Mtb2YtcHJlY2lzaW9uIjpbImVycm9yIl0sIm5vLW1pc2xlYWRpbmctY2hhcmFjdGVyLWNsYXNzIjpbImVycm9yIl0sIm5vLW1peGVkLXNwYWNlcy1hbmQtdGFicyI6WyJlcnJvciJdLCJuby1uZXctc3ltYm9sIjpbImVycm9yIl0sIm5vLW5vbm9jdGFsLWRlY2ltYWwtZXNjYXBlIjpbImVycm9yIl0sIm5vLW9iai1jYWxscyI6WyJlcnJvciJdLCJuby1vY3RhbCI6WyJlcnJvciJdLCJuby1wcm90b3R5cGUtYnVpbHRpbnMiOlsiZXJyb3IiXSwibm8tcmVkZWNsYXJlIjpbImVycm9yIl0sIm5vLXJlZ2V4LXNwYWNlcyI6WyJlcnJvciJdLCJuby1zZWxmLWFzc2lnbiI6WyJlcnJvciJdLCJuby1zZXR0ZXItcmV0dXJuIjpbImVycm9yIl0sIm5vLXNoYWRvdy1yZXN0cmljdGVkLW5hbWVzIjpbImVycm9yIl0sIm5vLXNwYXJzZS1hcnJheXMiOlsiZXJyb3IiXSwibm8tdGhpcy1iZWZvcmUtc3VwZXIiOlsiZXJyb3IiXSwibm8tdW5kZWYiOlsiZXJyb3IiXSwibm8tdW5leHBlY3RlZC1tdWx0aWxpbmUiOlsiZXJyb3IiXSwibm8tdW5yZWFjaGFibGUiOlsiZXJyb3IiXSwibm8tdW5zYWZlLWZpbmFsbHkiOlsiZXJyb3IiXSwibm8tdW5zYWZlLW5lZ2F0aW9uIjpbImVycm9yIl0sIm5vLXVuc2FmZS1vcHRpb25hbC1jaGFpbmluZyI6WyJlcnJvciJdLCJuby11bnVzZWQtbGFiZWxzIjpbImVycm9yIl0sIm5vLXVudXNlZC12YXJzIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtYmFja3JlZmVyZW5jZSI6WyJlcnJvciJdLCJuby11c2VsZXNzLWNhdGNoIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtZXNjYXBlIjpbImVycm9yIl0sIm5vLXdpdGgiOlsiZXJyb3IiXSwicmVxdWlyZS15aWVsZCI6WyJlcnJvciJdLCJ1c2UtaXNuYW4iOlsiZXJyb3IiXSwidmFsaWQtdHlwZW9mIjpbImVycm9yIl0sIm5vLXVzZS1iZWZvcmUtZGVmaW5lIjpbImVycm9yIl19fX0=)

**Comment by NullVoxPopuli** [comment]
ah! so the eslint rule is wrong!

**Comment by patricklx** [comment]
I looks like that's actually the behaviour it intended to do:

> This rule will warn when it encounters a reference to an identifier that has not yet been declared.

https://eslint.org/docs/latest/rules/no-use-before-define#rule-details

**Comment by NullVoxPopuli** [comment]
yeah -- but .. it's safe!

**Comment by patricklx** [comment]
i think this can be closed then? as the rule does work correctly as per its own definition

**Comment by NullVoxPopuli** [comment]
It works incorrectly for how we need it to tho 🤔
